app/Models/User.php

```diff
@@ -115,6 +115,26 @@ public function hasAccessToRepository(Repository $repository): bool
         return $this->repositories()->where('repositories.id', $repository->id)->exists();
     }
 
+    public function hasAccessToPackage(Package $package): bool
+    {
+        // Users with UNSCOPED permission have access to all packages
+        if ($this->can(Permission::UNSCOPED)) {
+            return true;
+        }
+
+        // Otherwise, check repository-level access
+        // Load repository if not already loaded to avoid lazy loading violation
+        $repository = $package->relationLoaded('repository')
+            ? $package->repository
+            : Repository::find($package->repository_id);
```

Comment:
Use `$package->repository` instead. Eager loading will take care of everything else.